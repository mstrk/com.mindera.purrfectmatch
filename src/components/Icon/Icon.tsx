import Svg, {Path} from "react-native-svg";

export type Props = {
  name: keyof typeof IconPaths;
  size?: number;
  color?: string;
  testID?: string;
};

export function Icon({name, size = 32, color = "#434141", ...rest}: Props) {
  return (
    <Svg width={size} height={size} viewBox="0 0 32 32" fill="none" {...rest}>
      {IconPaths[name](color)}
    </Svg>
  );
}

const IconPaths = {
  unmatch: (color: string) => (
    <>
      <Path d="M24 8L8 24" stroke={color} strokeWidth="2" strokeLinecap="round" strokeLinejoin="round" />
      <Path d="M8 8L24 24" stroke={color} strokeWidth="2" strokeLinecap="round" strokeLinejoin="round" />
    </>
  ),
  match: (color: string) => (
    <Path
      d="M27.7867 6.14667C27.1057 5.46534 26.2971 4.92486 25.4071 4.55611C24.5172 4.18736 23.5633 3.99757 22.6 3.99757C21.6367 3.99757 20.6828 4.18736 19.7929 4.55611C18.9029 4.92486 18.0943 5.46534 17.4133 6.14667L16 7.56L14.5867 6.14667C13.2111 4.77108 11.3454 3.99828 9.4 3.99828C7.45463 3.99828 5.58893 4.77108 4.21334 6.14667C2.83775 7.52226 2.06495 9.38795 2.06495 11.3333C2.06495 13.2787 2.83775 15.1444 4.21334 16.52L5.62667 17.9333L16 28.3067L26.3733 17.9333L27.7867 16.52C28.468 15.839 29.0085 15.0304 29.3772 14.1405C29.746 13.2505 29.9358 12.2966 29.9358 11.3333C29.9358 10.37 29.746 9.41614 29.3772 8.52619C29.0085 7.63625 28.468 6.82767 27.7867 6.14667V6.14667Z"
      fill={color}
      stroke={color}
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  ),
  paw: (color: string) => (
    <>
      <Path
        d="M17.1939 17.3473C21.8816 18.8823 27.2215 24.837 22.6125 28.5444C20.2678 30.4318 18.4572 28.788 16.1893 27.8603C13.4115 26.7263 11.839 28.61 9.08939 28.3007C7.25819 28.0945 5.59943 27.3936 4.84033 25.6786C2.65301 20.7285 11.033 17.0342 15.0253 16.978C15.7019 16.9686 16.4386 17.0999 17.1939 17.3473Z"
        fill={color}
      />
      <Path
        d="M25.0716 19.6227C26.3686 19.9207 27.8268 19.7127 28.8483 18.8598C29.8698 18.007 30.3291 16.457 29.7349 15.2668C28.719 13.235 26.3742 13.9604 25.1409 15.2555C24.5599 15.8647 24.0576 16.605 23.9695 17.4428C23.8795 18.2807 25.0716 19.6227 25.0716 19.6227Z"
        fill={color}
      />
      <Path
        d="M25.2928 8.87162C25.514 7.76577 25.6227 6.58497 25.096 5.54473C24.5693 4.50448 23.2235 3.67979 21.9528 3.95344C20.845 4.19148 20.174 5.1305 19.7129 6.02454C19.25 6.91859 18.8808 7.84825 18.6071 8.8004C18.311 9.83502 18.1385 10.9652 18.6465 11.9511C20.8169 16.1683 24.8711 10.9877 25.2928 8.87162Z"
        fill={color}
      />
      <Path
        d="M15.4339 11.4132C16.8172 9.68506 17.3195 7.47525 16.7741 5.50347C16.4854 4.45198 15.8557 3.42298 14.7629 2.91317C10.1221 0.755838 9.2206 9.96622 11.8465 11.7262C12.9955 12.4965 15.4339 11.4132 15.4339 11.4132Z"
        fill={color}
      />
      <Path
        d="M7.19819 13.8929C8.33215 12.3279 7.76424 9.21277 6.60967 7.79204C6.11297 7.17914 5.39889 6.65993 4.6098 6.69367C3.50396 6.7424 2.74111 7.82952 2.3625 8.86976C1.57529 11.0346 1.92014 14.5283 4.84781 14.8113C5.9724 14.92 6.72961 14.5377 7.19819 13.8929Z"
        fill={color}
      />
    </>
  ),
  chat: (color: string) => (
    <Path
      d="M28 15.3334C28.0046 17.0932 27.5934 18.8292 26.8 20.4C25.8592 22.2824 24.413 23.8656 22.6233 24.9724C20.8335 26.0792 18.771 26.6659 16.6667 26.6667C14.9068 26.6713 13.1708 26.2601 11.6 25.4667L4 28L6.53333 20.4C5.73991 18.8292 5.32874 17.0932 5.33333 15.3334C5.33415 13.229 5.92082 11.1665 7.02762 9.37677C8.13443 7.58704 9.71767 6.14079 11.6 5.20004C13.1708 4.40661 14.9068 3.99545 16.6667 4.00004H17.3333C20.1125 4.15336 22.7374 5.32639 24.7055 7.29452C26.6736 9.26265 27.8467 11.8876 28 14.6667V15.3334Z"
      stroke={color}
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  ),
  user: (color: string) => (
    <>
      <Path
        d="M26.6667 28V25.3333C26.6667 23.9188 26.1048 22.5623 25.1046 21.5621C24.1044 20.5619 22.7478 20 21.3333 20H10.6667C9.25217 20 7.89562 20.5619 6.89542 21.5621C5.89523 22.5623 5.33333 23.9188 5.33333 25.3333V28"
        stroke={color}
        strokeWidth="2"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <Path
        d="M16 14.6667C18.9455 14.6667 21.3333 12.2789 21.3333 9.33333C21.3333 6.38781 18.9455 4 16 4C13.0545 4 10.6667 6.38781 10.6667 9.33333C10.6667 12.2789 13.0545 14.6667 16 14.6667Z"
        stroke={color}
        strokeWidth="2"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </>
  ),
};
