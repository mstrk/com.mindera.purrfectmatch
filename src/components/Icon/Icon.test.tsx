import {render} from "@testing-library/react-native";
import {Icon} from "./Icon";

describe("Icon Component", () => {
  it("renders correctly with default props", () => {
    const {toJSON} = render(<Icon name="match" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders correctly with custom size and color", () => {
    const {toJSON} = render(<Icon name="unmatch" size={48} color="#FF0000" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders the correct icon based on the name prop", () => {
    const testID = "paw-icon";
    const {getByTestId} = render(<Icon name="paw" testID={testID} />);
    const svgElement = getByTestId(testID);
    expect(svgElement).toBeTruthy();
  });

  it("applies the correct size to the SVG element", () => {
    const testID = "chat-icon";
    const {getByTestId} = render(<Icon name="chat" size={64} testID={testID} />);
    const svgElement = getByTestId(testID);
    expect(svgElement.props.width).toBe(64);
    expect(svgElement.props.height).toBe(64);
  });
});
