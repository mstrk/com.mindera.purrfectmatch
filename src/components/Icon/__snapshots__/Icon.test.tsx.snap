// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Component renders correctly with custom size and color 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight={48}
  bbWidth={48}
  fill="none"
  focusable={false}
  height={48}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 48,
        "width": 48,
      },
    ]
  }
  vbHeight={32}
  vbWidth={32}
  width={48}
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
      ]
    }
  >
    <RNSVGPath
      d="M24 8L8 24"
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "stroke",
          "strokeWidth",
          "strokeLinecap",
          "strokeLinejoin",
        ]
      }
      stroke={
        {
          "payload": 4294901760,
          "type": 0,
        }
      }
      strokeLinecap={1}
      strokeLinejoin={1}
      strokeWidth="2"
    />
    <RNSVGPath
      d="M8 8L24 24"
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "stroke",
          "strokeWidth",
          "strokeLinecap",
          "strokeLinejoin",
        ]
      }
      stroke={
        {
          "payload": 4294901760,
          "type": 0,
        }
      }
      strokeLinecap={1}
      strokeLinejoin={1}
      strokeWidth="2"
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`Icon Component renders correctly with default props 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight={32}
  bbWidth={32}
  fill="none"
  focusable={false}
  height={32}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 32,
        "width": 32,
      },
    ]
  }
  vbHeight={32}
  vbWidth={32}
  width={32}
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
      ]
    }
  >
    <RNSVGPath
      d="M27.7867 6.14667C27.1057 5.46534 26.2971 4.92486 25.4071 4.55611C24.5172 4.18736 23.5633 3.99757 22.6 3.99757C21.6367 3.99757 20.6828 4.18736 19.7929 4.55611C18.9029 4.92486 18.0943 5.46534 17.4133 6.14667L16 7.56L14.5867 6.14667C13.2111 4.77108 11.3454 3.99828 9.4 3.99828C7.45463 3.99828 5.58893 4.77108 4.21334 6.14667C2.83775 7.52226 2.06495 9.38795 2.06495 11.3333C2.06495 13.2787 2.83775 15.1444 4.21334 16.52L5.62667 17.9333L16 28.3067L26.3733 17.9333L27.7867 16.52C28.468 15.839 29.0085 15.0304 29.3772 14.1405C29.746 13.2505 29.9358 12.2966 29.9358 11.3333C29.9358 10.37 29.746 9.41614 29.3772 8.52619C29.0085 7.63625 28.468 6.82767 27.7867 6.14667V6.14667Z"
      fill={
        {
          "payload": 4282597697,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
          "strokeLinecap",
          "strokeLinejoin",
        ]
      }
      stroke={
        {
          "payload": 4282597697,
          "type": 0,
        }
      }
      strokeLinecap={1}
      strokeLinejoin={1}
      strokeWidth="2"
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
